src/Model/FileStorage.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Kirby\Model;
+
+use Kirby\Data\Data;
+use Kirby\Exception\DuplicateException;
+use Kirby\Filesystem\F;
+use Kirby\Uuid\Uuid;
+use Throwable;
+
+class FileStorage extends Storage
+{
+	public function changeFilename(FileMeta $oldMeta, FileMeta $newMeta): void
+	{
+		$name      = F::name($newMeta->filename);
+		$extension = F::extension($newMeta->filename);
+		$directory = dirname($newMeta->identifier);
+
+		$newMeta->identifier = $directory . '/' . $name . '.' . $extension;
+
+		$this->moveIdentifier($oldMeta, $newMeta);
+	}
+
+	public function changeNum(FileMeta $oldMeta, FileMeta $newMeta): void
+	{
+		$this->update([
+			'num' => $newMeta->num,
+		]);
+	}
+
+	public function changeTemplate(FileMeta $oldMeta, FileMeta $newMeta): void
+	{
+		$this->update([
+			'template' => $newMeta->template,
+		]);
+	}
+
+	protected static function contentFile($meta): string
+	{
+		return $meta->identifier . '.txt';
+	}
+
+	public static function create(FileMeta $meta): File
+	{
+		// create the new identifier
+		$meta->identifier = static::createIdentifier($meta);
+
+		// generate the new UUID if it is not set yet
+		$meta->uuid ??= Uuid::generate();
```

Comment:
`Thought`: We need to find a good place for such code that isn't actually dependent on the storage handler. A database handler shouldn't have to reimplement things like UUID generation.